api/src/main/scala/org/virtuslab/ideprobe/protocol/ApplicationRunConfiguration.scala

```diff
@@ -13,6 +13,12 @@ final case class JUnitRunConfiguration(
     directory: Option[String]
 )
 
+final case class ScalaTestRunConfiguration(module: ModuleRef,
+                                           packageName: Option[String],
+                                           className: Option[String],
+                                           testName: Option[String],
```

Comment:
It seems that the three optional fields in this class are intended to be mutually exclusive. In that case, a sum type (a sealed trait with extending case classes) would be a much better fit.
```scala
sealed abstract class ScalaTestRunConfiguration(val moduleRef: ModuleRef)

case class PackageConfiguration(override val moduleRef: ModuleRef, packageName: String) extends ScalaTestRunConfiguration(moduleRef)
case class ClassConfiguration(override val moduleRef: ModuleRef, className: String) extends ScalaTestRunConfiguration(moduleRef)
case class TestConfiguration(override val moduleRef: ModuleRef, testName: String) extends ScalaTestRunConfiguration(moduleRef)
```

Reply:
I think TestConfiguration should have both `className` and `testName`. Initially I took this approach for junit when we had a different json serializer without good support of this. It would be generally better for the code to use ADT. I am hoping we can do this refactor along with a correct implementaton of running tests (hooking in to run configuration producers), so that it can be do once.